### ISSUE TITLE
do/while loops

### ISSUE THREAD

**Issue by liquibyte** [frang75/nappgui_src#133]
Why is this?  "do/while. Not allowed. Use for or while."  Seems like you're missing a third of available loops here.  I have a function that just doesn't work right without an exit-controlled loop.  Can you explain how I would rewrite a do/while loop using only while or for loops?

For example:
do {
    dest += dest_step;
    memmove(dest, src, img);
} while (--zmag > 0);

I've tried:
while (--zmag > 0) {
     dest += dest_step;
    memmove(dest, src, img);
}
Alas, it just doesn't perform the same.

I'm not a savant at this so I may be missing something rather obvious to those schooled in such matters.

**Comment by frang75** [comment]
These are just my standards when programming NAppGUI. You simply code however you want.
Please, this is a place to report ideas or bugs about the SDK. Not to open absurd discussions.

**Comment by liquibyte** [comment]
I didn't mean for it to come off as an absurd discussion.  My reading of that statement was that if I tried to port over my code to use your toolkit that I would somehow get an error if I tried to use do/while loops.  I wasn't aware that you were just stating your preference in the "Use of C" section under tutorial.

while. Nothing to comment.
do/while. Not allowed. Use for or while.
for. For infinite loops, use for(;;) instead of while(TRUE), as it avoids warnings in some compilers. Since there are ANSI-C based compilers, such as MSVC++ 8.0, we do not use variable declarations inside the for(), a feature that was introduced in C99.

These lines led me to believe, wrongly on my part apparently, that do/while was somehow forbidden while using NAppGUI.  I appologize for the misunderstanding and I sincerly hope that my original post didn't come across as absurd.
